sktime/tests/test_softdeps.py

```diff
@@ -325,3 +326,53 @@ def test_est_fit_without_modulenotfound(estimator):
             f'to the "python_dependencies" tag, and python version bounds should be'
             f' added to the "python_version" tag. Exception text: {error_msg}'
         ) from e
+
+
+@patch("sktime.utils.dependencies._dependencies.sys")
+@pytest.mark.parametrize(
+    "mock_release_version, prereleases, expect_exception",
+    [
+        (True, True, False),
+        (True, False, True),
+        (False, False, False),
+        (False, True, False),
+    ],
+)
+def test_check_python_version(
+    mock_sys, mock_release_version, prereleases, expect_exception
+):
+    from sktime.base import BaseObject
+
+    if mock_release_version:
+        mock_sys.version = "3.8.1rc"
+    else:
+        mock_sys.version = "3.8.1"
+
+    class DummyObjectClass(BaseObject):
+        _tags = {
+            "python_version": ">=3.7.1",  # PEP 440 version specifier, e.g., ">=3.7"
+            "python_dependencies": None,  # PEP 440 dependency strs, e.g., "pandas>=1.0"
+            "env_marker": None,  # PEP 508 environment marker, e.g., "os_name=='posix'"
+        }
+        """Define dummy class to test set_tags."""
+
+    dummy_object_instance = DummyObjectClass()
+
+    try:
+        _check_python_version(dummy_object_instance, prereleases=prereleases)
+    except ModuleNotFoundError as exception:
+        expected_msg = (
+            f"{type(dummy_object_instance).__name__} requires python version "
+            "to be {dummy_object_instance._tags['python_version']}, "
+            "but system python version is {mock_sys.version}. "
+            "This is due to the release candidate status of your system Python."
```

Comment:
```suggestion
            f"{type(dummy_object_instance).__name__} requires python version "
            f"to be {dummy_object_instance._tags['python_version']}, "
            f"but system python version is {mock_sys.version}. "
            "This is due to the release candidate status of your system Python."
```

@Abelarm please confirm that these are expected to be f-strings only. I am worried why are tests passing without `f`'s. Any ideas @fkiraly if some CI logic is disabling these tests?

Reply:
the tests should be triggered - which rn passed?

Reply:
No there was an error in the if statement making the test always passing even with the wrong string.
Now it's fixed!